build-support/bin/generate_github_workflows.py

```diff
@@ -545,8 +546,38 @@ def upload_log_artifacts(self, name: str) -> Step:
             "if": "always()",
             "continue-on-error": True,
             "with": {
-                "name": f"pants-log-{name.replace('/', '_')}-{self.platform_name()}",
-                "path": ".pants.d/pants.log",
+                "name": f"logs-{name.replace('/', '_')}-{self.platform_name()}",
+                "path": ".pants.d/*.log",
+            },
+        }
+
+    def upload_test_reports(self) -> Step:
+        date_str = datetime.now(timezone.utc).date().isoformat()
+        # The path doesn't include job ID, as we want to aggregate test reports across all
+        # jobs/shards in a workflow.  We do, however, qualify by run attempt, so we capture
+        # separate reports for tests that flake between attempts on the same workflow run.
+        s3_path = (
+            "test/reports/"
+            + self.platform_name()
+            + "/"
+            + date_str
```

Comment:
Yes, there was no way I could find to get the workflow run timestamp. 

But I think this is OK? I imagine the most common use will be to look at a set of files for the same test over several days, so pinpointing is probably not that important? And in any case, you can always find the data for a specific PR/branch by globbing over the date.

Reply:
Hm, if we want a low cardinality time-like grouping, the irregularly updated workflow generator rerun date could potentially be switched to "most recent git tag" or something, so that the generator is reproducible, at least.

Some almost time-of-workflow-run options:

1. the time of upload (ie `$(date ...)`). downside: a single run might be categorised in multiple days, if it gets unlucky and straddles the boundary 
2. the timestamp of the HEAD commit (maybe `$(git show --no-patch --format=%cd --date=format:%Y-%m-%d)`), similar to https://github.com/pantsbuild/pants/issues/7399#issuecomment-480622373 / https://github.com/pantsbuild/pants/pull/19179 . Downside: potentially inaccurate timestamp for a delayed rerun

I'd personally lean towards 2: that downside seems less problematic than 1's, and using usually-actual time seems more helpful than git tags or whatever.

Reply:
Ugh, I realized on second reading of your comment that my original code was a stupid brain-fart, I got confused between generator runtime and copy_to_s3.py runtime.  Anyway, will use your git invocation.